distributed/dashboard/components/scheduler.py

```diff
@@ -1721,6 +1730,450 @@ def __del__(self):
+                                <span style="font-size: 10px; font-family: Monaco, monospace;">@in_processing</span>
+                            </div>
+
+                            <div style="margin-left: 2em;">
+                                <span style="font-size: 12px; font-weight: bold;">In memory:</span>&nbsp;
+                                <span style="font-size: 10px; font-family: Monaco, monospace;">@in_memory</span>
+                            </div>
+
+                            <div style="margin-left: 2em;">
+                                <span style="font-size: 12px; font-weight: bold;">Erred:</span>&nbsp;
+                                <span style="font-size: 10px; font-family: Monaco, monospace;">@in_erred</span>
+                            </div>
+
+                            <div style="margin-left: 2em;">
+                                <span style="font-size: 12px; font-weight: bold;">Released:</span>&nbsp;
+                                <span style="font-size: 10px; font-family: Monaco, monospace;">@in_released</span>
+                            </div>
+                            """,
+            renderers=[rect],
+        )
+
+        self.root.add_tools(hover)
+
+    @without_property_validation
+    def update_layout(self):
+
+        with log_errors():
+            # get dependecies per task group
+            # in some cases there are tg that have themeselves as dependencies, we remove those.
+            dependencies = {
+                k: {ds.name for ds in ts.dependencies if ds.name != k}
+                for k, ts in self.scheduler.task_groups.items()
+            }
+
+            import dask
+
+            order = dask.order.order(
+                dsk={group.name: 1 for k, group in self.scheduler.task_groups.items()},
+                dependencies=dependencies,
+            )
+
+            ordered = sorted(self.scheduler.task_groups, key=order.get)
+
+            xs = {}
+            ys = {}
+            locations = set()
+            nodes_layout = {}
+            arrows_layout = {}
+            for tg in ordered:
+                if dependencies[tg]:
+                    x = max(xs[dep] for dep in dependencies[tg]) + 1
+                    y = max(ys[dep] for dep in dependencies[tg])
+                    if (
+                        len(dependencies[tg]) > 1
+                        and len({ys[dep] for dep in dependencies[tg]}) == 1
+                    ):
+                        y += 1
+                else:
+                    x = 0
+                    y = max(ys.values()) + 1 if ys else 0
+
+                while (x, y) in locations:  # avoid collisions by moving up
+                    y += 1
+
+                locations.add((x, y))
+
+                xs[tg], ys[tg] = x, y
+
+                # info neded for node layout to coulmn data source
+                nodes_layout[tg] = {}
+                nodes_layout[tg]["x"] = xs[tg]
+                nodes_layout[tg]["y"] = ys[tg]
```

Comment:
Maybe a little simpler

```suggestion
                nodes_layout[tg] = {"x": xs[tg], "y": ys[tg]}
```